src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -550,4 +553,31 @@ private void checkForConnectedDevisesConflict(String flowId, SwitchId switchId)
             }
         }
     }
+
+    private void validateQinQonWB(RequestedFlow requestedFlow)
+            throws UnavailableFlowEndpointException, InvalidFlowException {
+        if (requestedFlow.getSrcInnerVlan() != 0) {
```

Comment:
`requestedFlow.getSrcInnerVlan() != 0 && requestedFlow.getSrcVlan() != 0`
Otherwise inner vlan will just become vlan and this should be allowed

Reply:
Do we need a flow encapsulation check? Since there is no defect with vxlan and wb on the latest firmware

Reply:
Fixed case when inner VLAN != 0 and outer VLAN = 0.